docs/proposals/1374-multi-cluster-inference/README.md

```diff
@@ -0,0 +1,622 @@
+an implementation MAY use EPP metrics and/or health data to form a load-balancing decision.
+
+
+## Alternatives
+
+### Option 1: Reuse MCS API for EPP
+
+Reuse MCS to export EPP Services. This approach provides simple infra, but may be confusing to users (you “export EPPs” not pools) and
+requires a separate MCS parent export for parent-based inter-cluster routing.
+
+**Pros**:
+
+- Reuses existing MCS infrastructure.
+- Relatively simple to implement.
+
+**Cons**:
+
+- Referencing InferencePools in other clusters requires you to create an InferencePool locally.
+- In this model, you don’t actually choose to export an InferencePool, you export the EPP or InferencePool parent(s) service, that could lead to significant confusion.
+- InferencePool is meant to be a replacement for a Service so it may seem counterintuitive for a user to create a Service to achieve multi-cluster inference.
+
+## Option 2: New MCS API
+
+One of the key pain points we’re seeing here is that the current iteration of the MCS API requires a tight coupling between name/namespace and kind, with Service being the only kind of backend supported right now. This goes against the broader SIG-Network direction of introducing more focused kinds of backends (like InferencePool). To address this, we could create a resource that has an `exportRef` that allows for exporting different types of resources.
+
+While we were at it, we could combine the separate `export` and `import` resources that exist today, with `export` acting as the (optional) spec of this new resource, and `import` acting as `status` of the resource. Instead of `import` resources being automatically created, users would create them wherever they wanted to reference or export something to a MultiClusterService.
+
+Here’s a very rough example:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: MultiClusterService
+metadata:
+  name: bookinfo
+  namespace: bookinfo
+spec:
+  exportRef:
+    group: v1
+    kind: Service
+    name: bookinfo
+    scope: ClusterSet
+status:
+  conditions:
+  - type: Accepted
+    status: "True"
+    message: "MultiClusterService has been accepted"
+    lastTransitionTime: "2025-03-30T01:33:51Z"
+  targetCount: 1
+  ports:
+  - protocol: TCP
+    appProtocol: HTTP
+    port: 8080
+```
+
+### Open Questions
+
+#### Export Semantics
+
+- Finalize annotation key/values and ClusterSet discovery. Should we introduce a first-class exportPolicy field later?
+- Should we allow per-cluster export configuration (weights, region, SLO tags) at export time?
+
+#### InferencePool status surface
+
+- Do we extend InferencePool.status to publish EPP Service type/addresses/ports, health, metrics, and optional secret refs to simplify imports?
+- Should EPP Deployment/Pod discovery be standardized (labels/port names) for health/metrics auto-discovery?
+
+#### Security
+
+- Standard way to bootstrap mTLS between importing IG and exported EPP/parents, e.g. use BackendTLSPolicy?
```

Comment:
BTLSPolicy will not do mTLS. I strongly recommend using a service mesh here instead of trying to re-invent one within GIE. I promise its harder than it looks to do multi-cluster mTLS (https://blog.howardjohn.info/posts/mtls-kubernetes/#do-it-yourself) 🙂

Reply:
I mean a Gateway can specify a client cert, and BackendTLSPolicy can be used to validate the cert served by a backend (EPP in this case), so presumably mTLS wouldn't be that much of a stretch, even if mTLS provided by a mesh would be significantly easier to work with.